### ISSUE TITLE
S20 - Expansion by AM2301 Temperature &  Humidity

### ISSUE THREAD

**Issue by FabianHM** [arendst/Sonoff-Tasmota#670]
Simple solution to expand the S20 by an AM2301 module.
Required:
- Jack socket 3.5 mm stereo
- AM2301 (DHT21) Digital Temperatur& Humidity Sensor
- 0.5m audio cable  with 3.5mm stereo plug
![2017-07-31 1](https://user-images.githubusercontent.com/26110738/28793796-7d44407e-7634-11e7-9735-8777e3c7b105.jpg)
![2017-07-31 2](https://user-images.githubusercontent.com/26110738/28793813-8b6c1370-7634-11e7-8b8e-c42afff512ea.jpg)
![2017-07-31 3](https://user-images.githubusercontent.com/26110738/28793828-95fb6282-7634-11e7-931b-22e55cdb8a14.jpg)
![2017-07-31 4](https://user-images.githubusercontent.com/26110738/28793831-97d430a2-7634-11e7-8cf2-99cf73c6e842.jpg)

**Comment by jat80** [comment]
What pins go to what from the S20 to the audo jack?

**Comment by arendst** [comment]
Thnx. Closed issue and add link in Wiki.

**Comment by jat80** [comment]
@arendst I still don't know how to configure this?

**Comment by arendst** [comment]
For others:

![image](https://user-images.githubusercontent.com/11044339/29211691-cd328a1e-7e9a-11e7-93a3-a48e3490fed5.png)

**Comment by jat80** [comment]
@arendst It's the wiring for the jack to correspond with the socket that would be the issue for most people.

**Comment by FabianHM** [comment]
You need only connect 3 cables!
AM2301 red to S20 VCC (3.3V)
AM2301 black to S20 GND (Ground)
AM2301 yellow to S20 RX or TX (GPIO 1 or 3)

**Comment by jat80** [comment]
@FabianHM how would someone wire the 3.5mm jack is what i'm saying.  the above is obvious but if someone wants to make there own jack to the sensor themselves, which pins correspond with the jack pins - jack to jack socket.

**Comment by FabianHM** [comment]
Connect the audio cable with 3.5mm stereo plug with the Jack socket.
Now simply measure the pin assignment of the socket with a measuring device. The allocation of the bushes are different in the case of some manufacturers.

**Comment by CaptInsano** [comment]
Is any pull-up resistor needed between VCC and DATA?

I am getting elevated Temp measurements from my AM2301s (I have tested with 3 different AM2301s) on different S20 sockets and I am trying to debug why.

I have collected a DS18B20 to the other S20 pin to compare readings, see below:

![capture](https://user-images.githubusercontent.com/332478/30786473-199fe824-a16e-11e7-8e0c-6d24c3572059.PNG)

![capture1](https://user-images.githubusercontent.com/332478/30786477-33b83e28-a16e-11e7-9cd9-b1f67e29743f.PNG)

Any ideas?
